### ISSUE TITLE
imagec not displaying all of the images it's grabbing

### ISSUE THREAD

**Issue by jzt** [vmware/vic#524]
**VIC Version:**

0.1 / master / 4b9728bc794c51070ae4ea9131bad0372ef5535a

**Code**

Using docker:

```
vagrant@vagrant:~$ docker pull tomcat
Using default tag: latest
latest: Pulling from library/tomcat
70e9a6907f10: Pull complete
32f2a4cccab8: Pull complete
004814f54a9a: Pull complete
e37211b690fb: Pull complete
01c48606610e: Pull complete
d792bfd7ef56: Pull complete
60aa2c08d599: Pull complete
aa12b2ef318b: Pull complete
2b19c26dffa8: Pull complete
63db1d3e3c2a: Pull complete
5f20fd3dd28c: Pull complete
f9989a3fd068: Pull complete
020991c79298: Pull complete
c742cd313dd7: Pull complete
4cd64b60fdc9: Pull complete
4242b893da9f: Pull complete
a725617cbc90: Pull complete
daa283ff188d: Pull complete
144cc457ebd7: Pull complete
b03113766210: Pull complete
4b841eee8613: Pull complete
Digest: sha256:d2eeba2fc1da6d9a092bd103b69b527196155568b723eb67e0355033f3a3af29
Status: Downloaded newer image for tomcat:latest
```

Using vic:

```
Using vic:

vagrant@vagrant:~/go/src/github.com/vmware/vic$ DOCKER_HOST=tcp://localhost:2376 docker --tls --tlscert='cert.pem' --tlskey='key.pem' pull tomcat
Using default tag: latest
latest: Pulling from library/tomcat
a3ed95caeb02: Pull complete
efd26ecc9548: Pull complete
d1784d73276e: Pull complete
52a884c93bb2: Pull complete
c35f0a4a3a31: Pull complete
d01b68d70ee0: Pull complete
73faba584c67: Pull complete
b23bdd8f4c62: Pull complete
d4f8ecdef04d: Pull complete
Status: Downloaded newer image for library/tomcat:latest
```

They both pull 21 layers in total.

**Comment by mdubya66** [comment]
@jzt is this still an issue?

**Comment by anchal-agrawal** [comment]
@jzt Looks like VIC is pulling one extra layer now - `a3ed95caeb02`. For `ubuntu`:

Docker:
```
43db9dbdcb30: Pull complete 
2dc64e8f8d4f: Pull complete 
670a583e1b50: Pull complete 
183b0bfcd10e: Pull complete 
Digest: sha256:c6674c44c6439673bf56536c1a15916639c47ea04c3d6296c5df938add67b54b
Status: Downloaded newer image for ubuntu:latest
```

VIC:
```
a3ed95caeb02: Pull complete 
43db9dbdcb30: Pull complete 
2dc64e8f8d4f: Pull complete 
670a583e1b50: Pull complete 
183b0bfcd10e: Pull complete 
Digest: sha256:85af8b61adffea165e84e47e0034923ec237754a208501fce5dbeecbb197062c
Status: Downloaded newer image for library/ubuntu:latest
```

I can reproduce this while pulling `tomcat` as well. The only extra layer that VIC is pulling in both cases is `a3ed95caeb02`. I'm not sure what that is for.

**Comment by mdubya66** [comment]
@jzt please triage and re-assign if you aren't up to it :)

**Comment by jzt** [comment]
@anchal-agrawal Yeah, that is to be expected right now. The `a3ed` layer you are seeing is the "empty layer", which is basically a layer in which a command was run but the filesystem did not change.

In late June/early July, Docker switched to using schema v2 manifests, which omit the empty layer. The image metadata format is different, too, in that it does not include empty layer diffID's, but it includes the associated empty layer commands in the history. This is something we need to support eventually.

TL;DR:  It's not abnormal that a) we see the empty layer being pulled, and b) we see different image ID's than docker (as the image metadata format has changed).

It appears, aside from the empty layer being pulled, that the original problem is no longer an issue.

**Comment by jzt** [comment]
cc @mdubya66

**Comment by mhagen-vmware** [comment]
@jzt close this then?

**Comment by jzt** [comment]
Yeah 👍
